harness/determined/searcher/search_runner.py

```diff
@@ -49,94 +34,189 @@ def run(
                         )
-                        if e.trialExitedEarly.exitedReason is None:
+                        if event.trialExitedEarly.exitedReason is None:
                             raise RuntimeError(
                                 "trialExitedEarly event is invalid without exitedReason"
                             )
-                        request_id = uuid.UUID(e.trialExitedEarly.requestId)
-                        if e.trialExitedEarly.exitedReason in (
+                        request_id = uuid.UUID(event.trialExitedEarly.requestId)
+                        if event.trialExitedEarly.exitedReason in (
                             v1TrialExitedEarlyExitedReason.EXITED_REASON_INVALID_HP,
                             v1TrialExitedEarlyExitedReason.EXITED_REASON_INIT_INVALID_HP,
                         ):
                             self.search_method.searcher_state.trial_progress.pop(request_id, None)
                         elif (
-                            e.trialExitedEarly.exitedReason
+                            event.trialExitedEarly.exitedReason
                             == v1TrialExitedEarlyExitedReason.EXITED_REASON_UNSPECIFIED
                         ):
                             self.search_method.searcher_state.failures.add(request_id)
                         operations = self.search_method.on_trial_exited_early(
                             request_id,
                             exited_reason=ExitedReason._from_bindings(
-                                e.trialExitedEarly.exitedReason
+                                event.trialExitedEarly.exitedReason
                             ),
                         )
-                    elif e.validationCompleted:
+                    elif event.validationCompleted:
                         # duplicate completion accounting already performed by master
                         logging.info(
-                            f"validationCompleted({e.validationCompleted.requestId},"
-                            f" {e.validationCompleted.metric})"
+                            f"validationCompleted({event.validationCompleted.requestId},"
+                            f" {event.validationCompleted.metric})"
                         )
-                        request_id = uuid.UUID(e.validationCompleted.requestId)
-                        if e.validationCompleted.metric is None:
+                        request_id = uuid.UUID(event.validationCompleted.requestId)
+                        if event.validationCompleted.metric is None:
                             raise RuntimeError(
                                 "validationCompleted event is invalid without a metric"
                             )
                         operations = self.search_method.on_validation_completed(
                             request_id,
-                            e.validationCompleted.metric,
+                            event.validationCompleted.metric,
```

Comment:
i'm not sure if this is too late but i feel strongly that we need to be able to know the train length corresponding to the metric which is information available from the operation that generated this event. this is somewhat of a requirement for ease of use when it comes to multi-fidelity bayesian methods that suggest hparams with associated training durations and then updates the bayesian model over joint space of hparams and training duration.  it would also give us a way to bypass saving checkpoint for progress events by making sure we eventually update trialProgress for each validationCompleted event.

Reply:
We can do it but let's leave it outside the scope of this PR.